refinery/analysis_manager/tasks.py

```diff
@@ -336,6 +358,67 @@ def _run_galaxy_workflow(analysis_uuid):
         return
 
 
+def _run_tool_based_galaxy_workflow(analysis_uuid):
+
+    analysis = _get_analysis(analysis_uuid)
+    analysis_status = _get_analysis_status(analysis_uuid)
+    tool = _get_tool(analysis_uuid)
+
+    if not analysis_status.galaxy_import_task_group_id:
+        connection = analysis.galaxy_connection()
+        library_dict = connection.libraries.create_library(
+            name="Library for: {}".format(tool)
+        )
+        history_dict = connection.histories.create_history(
+            name="History for: {}".format(tool)
+        )
+
+        logger.debug("Starting analysis execution in Galaxy")
+
+        galaxy_import_tasks = [
+            _tool_based_galaxy_file_import.subtask(
+                (
+                    analysis_uuid,
+                    file_store_item_uuid,
+                    history_dict,
+                    library_dict,
+                )
+            ) for file_store_item_uuid in tool.get_input_file_uuid_list()
+            ]
+
+        galaxy_file_import_taskset = TaskSet(
+            tasks=galaxy_import_tasks
+        ).apply_async()
+
+        galaxy_file_import_taskset.save()
+
+        analysis_status.galaxy_import_task_group_id = (
+            galaxy_file_import_taskset.taskset_id
+        )
+        analysis_status.set_galaxy_history_state(AnalysisStatus.PROGRESS)
+        run_analysis.retry(countdown=RETRY_INTERVAL)
+
+    # Check if data files were successfully imported into Galaxy
+    galaxy_file_import_taskset = get_taskset_result(
+        analysis_status.galaxy_import_task_group_id
+    )
+    if not galaxy_file_import_taskset.ready():
+        logger.debug("Analysis '%s' pending in Galaxy", analysis)
+        run_analysis.retry(countdown=RETRY_INTERVAL)
+    elif not galaxy_file_import_taskset.successful():
```

Comment:
Are you sure this is reachable? I could imagine if something is not `successful` it might not be `ready` either.